wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+	}
+
+	/**
+	 * Initializes the admin tasks.
+	 */
+	public function init_admin() {
+		$this->register_settings();
+		$this->add_settings_field();
+	}
+
+	/**
+	 * Registers the HTTPS settings.
+	 */
+	public function register_settings() {
+		$args = array(
+			'type'              => 'string',
+			'sanitize_callback' => array( $this, 'sanitize_callback' ),
+		);
+
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_HTTPS_OPTION,
+			$args
+		);
+
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_INSECURE_CONTENT_OPTION,
+			$args
+		);
+	}
+
+	/**
+	 * Sanitizes the raw setting value.
+	 *
+	 * @param string $raw_setting The setting before sanitizing it.
+	 * @return string|null The setting, or null if it's not valid.
+	 */
+	public function sanitize_callback( $raw_setting ) {
+		if ( self::OPTION_SELECTED_VALUE === $raw_setting || '0' === $raw_setting ) {
+			return $raw_setting;
+		}
+	}
+
+	/**
+	 * Adds a settings field to the 'Reading Settings' page.
+	 */
+	public function add_settings_field() {
+		add_settings_field(
+			self::SETTING_ID,
+			__( 'HTTPS', 'pwa' ),
+			array( $this, 'render_settings' ),
+			self::OPTION_GROUP
+		);
+	}
+
+	/**
+	 * Renders the HTTPS settings in /wp-admin on the Reading Settings page.
+	 */
+	public function render_settings() {
+		$upgrade_https_value           = get_option( self::UPGRADE_HTTPS_OPTION );
+		$upgrade_insecure_content      = get_option( self::UPGRADE_INSECURE_CONTENT_OPTION );
+		$https_more_details            = sprintf(
+			'<a href="%s">%s</a>',
+			'https://make.wordpress.org/support/user-manual/web-publishing/https-for-wordpress/',
+			esc_html__( 'More details', 'pwa' )
+		);
+		$insecure_content_more_details = sprintf(
+			'<a href="%s">%s</a>',
+			'https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content',
+			esc_html__( 'More details', 'pwa' )
+		);
+
+		/**
+		 * Todo: change this description based on what insecure content is found.
+		 *
+		 * Have a separate description for when there's only passive insecure content.
+		 * And one for when there's active and passive insecure content.
+		 *
+		 * @see https://github.com/xwp/pwa-wp/issues/17#issuecomment-406188455
+		 */
+		/* Translators: %s: a link for more details */
+		$insecure_content_description = esc_html__( 'Your home page doesn’t contain insecure URLs. However, there may be URLs on other pages that could be blocked. %s', 'pwa' );
+
+		/*
+		 * Todo: change !== to === as this is only like this for development.
+		 * The WP_HTTPS_Detection doesn't work with my local SSL certificate.
+		 * Also, change $this->is_currently_https() to ! $this->is_currently_https().
+		 * This is also for development only.
+		 * This main if block should only run if the site can use HTTPS, but the 'home' and 'siteurl' options aren't HTTPS.
+		 */
+		if ( $this->is_currently_https() && true !== get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) ) :
```

Comment:
I believe that `get_option()` always returns a string when it is a scalar value, since `option_value` is a text field. So this would be better to just check if `get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME )` is truthy I believe.

Reply:
Thanks, [https://github.com/xwp/pwa-wp/pull/36/commits/cef3e592a5c8e837c77f2dd38984507c7dc4b953](https://github.com/xwp/pwa-wp/pull/36/commits/cef3e592a5c8e837c77f2dd38984507c7dc4b953) evaluates whether it's 'truthy.'